utils/categories.js

```diff
@@ -0,0 +1,90 @@
+const categories = [
+  "African",
+  "Asian",
+  "American",
+  "British",
+  "Cajun",
+  "Caribbean",
+  "Chinese",
+  "Eastern European",
+  "European",
+  "French",
+  "German",
+  "Greek",
+  "Indian",
+  "Irish",
+  "Italian",
+  "Japanese",
+  "Jewish",
+  "Korean",
+  "Latin American",
+  "Mediterranean",
+  "Mexican",
+  "Middle Eastern",
+  "Nordic",
+  "Southern",
+  "Spanish",
+  "Thai",
+  "Vietnamese",
+];
+
+function createCategoryNav(categories) {
+  // get parameters from url
+  const urlParams = new URLSearchParams(window.location.search);
+  const id = urlParams.get("id");
+
+  const maxNavItem = 9;
+  let categoryList = [];
+  const categoryNav = document.querySelector(".main-nav-bar");
+
+  categoryList =
+    categories?.length > maxNavItem
+      ? categories.slice(0, maxNavItem)
+      : categories;
+
+  categoryList.forEach((category) => {
+    // list item
+    const listItem = document.createElement("li");
+    listItem.classList.add("nav-item");
+
+    // link item
+    const linkItem = document.createElement("a");
+    linkItem.classList.add("nav-link");
+    if (!id) {
+      linkItem.addEventListener(
+        "click",
+        (ev) => getSelectedCategory(ev, category),
+        false
+      );
+    }
+
+    const categoryPath = !id ? "#" : `../index.html?category=${category}`;
```

Comment:
instead of "#" you can put `index.html?category=${category}` - so that in "outer" pages such as main page or category page the menu, as well as category names work correctly.

Reply:
done